carlsim/test/core.cpp

```diff
@@ -636,3 +636,18 @@ TEST(Core, saveLoadSimulation) {
 		}
 	}
 }
+
+TEST(Core, synapseIdOverflow) {
+	::testing::FLAGS_gtest_death_test_style = "threadsafe";
+
+	CARLsim sim("Core.synapseIdOverflow", HYBRID_MODE, SILENT, 0, 42);
+
```

Comment:
Is it that you have 16bits to address synapses, so having 2^16-1 synapses is fine, but starting at 2^16 it should break? If so, I would suggest creating 65,536 neurons in `gExc` instead of 80,000 to test the corner case.